src/ui/webeiddialog.cpp

```diff
@@ -127,6 +127,7 @@ WebEidDialog::WebEidDialog(QWidget* parent) : WebEidUI(parent), ui(new Private)
         showPinInputWarning(false);
         ui->okButton->setEnabled(ui->pinInput->hasAcceptableInput());
     });
+    connect(ui->pinInput, &QLineEdit::returnPressed, ui->okButton, &QPushButton::click);
```

Comment:
As the problem only exists on macOS, I would suggest conditional compilation:
```suggestion
#ifdef Q_OS_MAC
    connect(ui->pinInput, &QLineEdit::returnPressed, ui->okButton, &QPushButton::click);
#endif    
```

Reply:
@metsma, I'll add the conditional compilation myself.